apps/readest-app/src/app/reader/components/settings/ColorPanel.tsx

```diff
@@ -67,6 +84,23 @@ const ColorPanel: React.FC<{ bookKey: string }> = ({ bookKey }) => {
     );
   }, [settings]);
 
+  // manage syntax highlighting toggle
+  useEffect(() => {
+    const viewSettings = getViewSettings(bookKey);
+    const view = getView(bookKey);
+    if (!viewSettings || !view) return;
+    const docs = view.renderer.getContents();
+    // Apply or remove highlighting in all current documents based on the setting
+    for (const { doc } of docs) {
+      manageSyntaxHighlighting(doc, viewSettings, isDarkMode);
```

Comment:
It would be better to apply syntax highlighting directly in the `codeLanguage` and `codeLanguage` useEffect after `saveViewSettings` is called which will update the viewSettings.

Reply:
You can add `isDarkMode` in the dependencies of codeLanguage useEffect as `[codeLanguage, isDarkMode]`.